core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java

```diff
@@ -253,7 +253,13 @@ else if (element instanceof LikeClause) {
         Set<String> specifiedPropertyKeys = statement.getProperties().stream()
                 .map(property -> property.getName().getValue())
                 .collect(toImmutableSet());
-        Map<String, Object> finalProperties = combineProperties(specifiedPropertyKeys, properties, inheritedProperties);
+
+        Set<String> propertiesToSkip = plannerContext.getMetadata().getTablePropertiesToSkipForCreateTableLike(session, catalogName);
```

Comment:
not sure, but is it worth to check whether the skipped properties valid properties based on the the source table?

Reply:
By adding the flags on the property itself, it is now inherently validated.